### PR TITLE
Remove double BFE encode

### PR DESCRIPTION
**Pull request by mycognosist**
Since BFE encoding is recursive (for arrays and objects) but not idempotent, we need to ensure that objects are only encoded once. This is especially important for Buffers since their type-format identifiers get pushed out of position if double or triple encoded.

This PR bumps ssb-bfe to `3.0.1` and ensures single-encoding in `encodeNew()` and `validateSingle()`.

**Important** : tests using vectors are currently failing due to lack of SSB-URI support in the Go metafeed implementation (the source of the vector-generation code). This PR temporarily deactivates vector tests to allow for downstream development progress. Vector tests will be reinstated once up-to-date version are available.

### CODE REVIEW

File: test/testvector-metafeed-managment.json [hunk 1]
```diff
@@ -1,111 +1 @@
...
...74323a06026565343a7479706532333a06006d657461666565642f6164642f6578697374696e676536363a0400eef616ed1ba7a8a3ad15a9a4e3dcbe7c791346337fc1a7a5c206ad95a0ff504ae3a803390020d3c5f1ba2a50a6d6fcedb0fba99f1324d8a4253613f1a694730e656536363a040009bd0aa9b0618fabda1895f9496dc917b5e2580d37991e578628ee2e6e7a1dc12a5360e4793230fc6fd49a648aa56fd344d25a5238ac368ba9456de75e25760d65","Key":"%xgbmcNOUVRotdVxpqUcCJhC/90uNeusjrw4KANrFvOk=.bbmsg-v1","Author":"@b99R2e7lj8h7NFqGhOu6lCGy8gLxWV+J4ORd1X7rP3c=.bbfeed-v1","Sequence":4,"Previous":"%tirbHwMq4LX9MF5nI0bLVcdAK0YzptqOJNOiKGJR3pY=.bbmsg-v1","Timestamp":0,"HighlevelContent":[{"type":"metafeed/add/existing","feedpurpose":"metafeed upgrade of existing","subfeed":"@4x4183TbjTA46ROc5Uj9FmtE+H2bFVVeGjQzGwdlZCw=.bamboo","metafeed":"@b99R2e7lj8h7NFqGhOu6lCGy8gLxWV+J4ORd1X7rP3c=.bbfeed-v1","tangles":{"metafeed":{"root":null,"previous":null}}},{"Name":"subfeed signature","HexString":"0400eef616ed1ba7a8a3ad15a9a4e3dcbe7c791346337fc1a7a5c206ad95a0ff504ae3a803390020d3c5f1ba2a50a6d6fcedb0fba99f1324d8a4253613f1a694730e"}],"Signature":"040009bd0aa9b0618fabda1895f9496dc917b5e2580d37991e578628ee2e6e7a1dc12a5360e4793230fc6fd49a648aa56fd344d25a5238ac368ba9456de75e25760d"}]}
```
Comment (staltz) [inline test/testvector-metafeed-managment.json:1]: Could this JSON be formatted with the same newlines as it had before?

One programmatic way of achieving this is (in Node.js REPL):

```js
JSON.stringify(require('./test/testvector-metafeed-managment.json'), null, 2)
```

then copy-pasting that back into this json file

Comment (mycognosist) [reply test/testvector-metafeed-managment.json:1]: Ah yes, definitely! Thanks for pointing that out. I will need to push the new vector file once it's created and will be sure to pretty-print it beforehand.

**Comment by staltz** [comment]
(Asking because I'm not sure:) is this PR pending on vector files being updated AND ssb-bfe updated? Or both? (Thinking of putting a checklist at the top so we can keep track of what dominoes need to fall first)

**Comment by mycognosist** [comment]
This one is pending https://github.com/ssb-ngi-pointer/ssb-bfe/pull/29 and some test updates. I believe the vectors we currently have should be sufficient here.

**Comment by staltz** [comment]
The one case where I think our handwritten vector files won't work is the "EncodedData" which is binary encoded in hex :/

**Comment by mycognosist** [comment]
> The one case where I think our handwritten vector files won't work is the "EncodedData" which is binary encoded in hex :/

Indeed. We really need to wait for the updated vectors I think. But at least we've done a lot of the necessary changes (ssb-uri, ssb-bfe) so that the test updates will be quick once we have the latest vectors.

**Comment by mycognosist** [comment]
Note the **important** section in the original post.

**Comment by staltz** [comment]
@mycognosist If you bump the devDependency `ssb-meta-feeds` to 0.17.0 it might fix the npm install in CI.

**Review by staltz — APPROVED** [review]
:green_book:

**Pull request merged**